pulser-core/pulser/noise_model.py

```diff
@@ -237,13 +370,47 @@ def _check_eff_noise(self) -> None:
                     f"Operator's shape must be (2,2) not {operator.shape}."
                 )
 
+    @staticmethod
+    def _validate_parameters(param_vals: dict[str, Any]) -> None:
+        for param in param_vals:
+            is_valid = True
+            value = param_vals[param]
+            if param in _POSITIVE:
+                is_valid = value >= 0
+                comp = "greater than or equal to zero"
+            elif param in _STRICT_POSITIVE:
+                is_valid = value is not None and value > 0
+                comp = "greater than zero"
+            elif param in _PROBABILITY_LIKE:
+                is_valid = 0 <= value <= 1
+                comp = (
+                    "greater than or equal to zero and smaller than "
+                    "or equal to one"
+                )
+            if not is_valid:
+                raise ValueError(f"'{param}' must be {comp}, not {value}.")
+
     def _to_abstract_repr(self) -> dict[str, Any]:
         all_fields = asdict(self)
         eff_noise_rates = all_fields.pop("eff_noise_rates")
         eff_noise_opers = all_fields.pop("eff_noise_opers")
         all_fields["eff_noise"] = list(zip(eff_noise_rates, eff_noise_opers))
         return all_fields
 
+    def __repr__(self) -> str:
+        relevant_params = self._find_relevant_params(
+            self.noise_types,
+            self.state_prep_error,
+            self.amp_sigma,
+            self.laser_waist,
+        )
+        relevant_params.add("noise_types")
+        params_list = []
+        for f in fields(self):
+            if f.name in relevant_params:
+                params_list.append(f"{f.name}={getattr(self, f.name)!r}")
```

Comment:
Why not do:
```suggestion
       if param in relevant_params:
            params_list.append(f"{param}={getattr(self, param)!r}")
```

Reply:
Well I think that if you insert noise_types at 0, then everything is in order, but I agree it's surely not great to rely on that :)

Reply:
Not really because `relevant_params` is a set, so it just scrambles everything